src/shapes/Textbox.ts

```diff
@@ -302,22 +334,29 @@ export class Textbox extends IText {
 
     const data = lines.map((line, lineIndex) => {
       let offset = 0;
-      const words = splitByGrapheme
+      const wordsOrGraphemes = splitByGrapheme
         ? this.graphemeSplit(line)
         : this.wordSplit(line);
 
       // fix a difference between split and graphemeSplit
-      if (words.length === 0) {
-        words.push([]);
+      // why are we doing this? what difference?
+      // wouldn't it be better return data for empty words?
+      if (wordsOrGraphemes.length === 0) {
+        return {
+          word: '',
+          width: 0,
+        };
       }
 
-      return words.map((word) => {
+      return wordsOrGraphemes.map((word: string) => {
         // if using splitByGrapheme words are already in graphemes.
-        word = splitByGrapheme ? word : this.graphemeSplit(word);
-        const width = this._measureWord(word, lineIndex, offset);
+        const graphemeArray = splitByGrapheme
+          ? [word]
+          : this.graphemeSplit(word);
+        const width = this._measureWord(graphemeArray, lineIndex, offset);
         largestWordWidth = Math.max(width, largestWordWidth);
         offset += word.length + infix.length;
-        return { word, width };
+        return { word: graphemeArray, width };
```

Comment:
@ShaMan123 we change this function recently.
Word is a string, but then it becomes string or string[] ( because graphemeSplit returns string[];

GraphemeData is defined as wanting string, but definetely here is getting also string[];

I have tried to make this function more straight forward but i m blocked.

Reply:
I noticed it when working on it but was afraid to touch it
Your fix looks valid, change `GraphemeData` type to accept both but it can since it does `value[index]` only if I recall correctly

Reply:
but it doesn't pass type check :)